src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -377,19 +385,29 @@ SHAMap::descendAsync(
 
         if (!ptr && backed_)
         {
-            std::shared_ptr<NodeObject> obj;
-            if (!f_.db().asyncFetch(hash.as_uint256(), ledgerSeq_, obj))
+            std::shared_ptr<NodeObject> object;
+            if (f_.db().asyncFetch(hash.as_uint256(), ledgerSeq_, object,
```

Comment:
I'll note this is the only call to `asyncFetch`, which means all the calls use the same callback. We could consider removing the callback parameters and hard coding the callback - this makes the code easier to understand.

On the other hand, keeping the callbacks makes for a better interface to `Database` and keeps the components separate.

I'm in favour of keeping things as they are, but wanted to point this out in case other people had strong feelings of wanting to change this.